ush/python/pygfs/task/analysis.py

```diff
@@ -61,6 +67,47 @@ def get_obs_dict(self: Task) -> Dict[str, Any]:
         }
         return obs_dict
 
+    @logit(logger)
+    def get_bias_dict(self: Task) -> Dict[str, Any]:
+        """Compile a dictionary of observation files to copy
+
+        This method uses the OBS_LIST configuration variable to generate a dictionary
+        from a list of YAML files that specify what observation bias correction files
+        are to be copied to the run directory from the observation input directory
+
+        Parameters
+        ----------
+
+        Returns
+        ----------
+        obs_dict: Dict
+            a dictionary containing the list of observation files to copy for FileHandler
+        """
+        obs_list_config = YAMLFile(path=self.config['OBS_LIST'])
+        # get observers from master dictionary
+        observers = obs_list_config['observers']
+        copylist = []
+        for ob in observers:
+            if 'obs bias' in ob.keys():
+                obfile = ob['obs bias']['input file']
+                basename = os.path.basename(obfile)
+                copylist.append([os.path.join(self.task_config.comin_ges_atm, basename), obfile])
+
+                obfile2 = obfile.replace('satbias', 'satbias_cov')
+                basename = os.path.basename(obfile2)
+                copylist.append([os.path.join(self.task_config.comin_ges_atm, basename), obfile2])
+
+                obfile2 = obfile.replace('satbias', 'tlapse')
+                obfile2 = obfile2.replace('nc4', 'txt')
+                basename = os.path.basename(obfile2)
+                copylist.append([os.path.join(self.task_config.comin_ges_atm, basename), obfile2])
```

Comment:
This can be condensed to a single block:
```suggestion
               for obfile in ['satbias.nc4', 'satbias_cov.nc4', 'tlapse.txt']:
                    basename = os.path.basename(obfile)
                    copylist.append([os.path.join(self.task_config.comin_ges_atm, basename), obfile])
```
There might be some path that might need to be introduced, but I can't tell without knowing what `ob['obs bias']['input file']` is.

I might be able to do this better if I know what `obfile` is in that loop.

Reply:
Attempt to add suggestion at f196d3a9

Reply:
Not surprising my attempt to implement the intent of the pseudo code at f196d3a does not work.  

Below is the section of the yaml file `get_bias_dict` is processing in the test case
```
      obs bias:
        input file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.satbias.2021122018.nc4
        output file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/bc/gdas.t00z.amsua_n19.satbias.2021122100.nc4
        variational bc:
          predictors:
          - name: constant
          - name: lapse_rate
            order: 2
            tlapse: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.tlapse.2021122018.txt
          - name: lapse_rate
            tlapse: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.tlapse.2021122018.txt
          - name: emissivity
          - name: scan_angle
            order: 4
          - name: scan_angle
            order: 3
          - name: scan_angle
            order: 2
          - name: scan_angle
        covariance:
          minimal required obs number: 20
          variance range:
          - 1.0e-06
          - 10.0
          step size: 0.0001
          largest analysis variance: 10000.0
          prior:
            input file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.satbias_cov.2021122018.nc4
            inflation:
              ratio: 1.1
              ratio for small dataset: 2.0
          output file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/bc/gdas.t00z.amsua_n19.satbias_cov.2021122100.nc4
```
The lines we want to extract and process are 
```
input file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.satbias.2021122018.nc4

tlapse: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.tlapse.2021122018.txt

input file: /scratch2/NCEPDEV/stmp1/Russ.Treadon/RUNDIRS/prufsatm/gdasatmanl_00/obs/gdas.t18z.amsua_n19.satbias_cov.2021122018.nc4

```

Reply:
Commit revised `analysis.py` at 5e8cb1ce.   This update successfully runs `gdasatmanlinit`.   This update reverts to the `replace` method.   If existing `get_bias_dict` functionality can be achieved via a condensed code block, please guide me.  I don't know how to do this at present.